app/src/main/java/com/vci/vectorcamapp/core/data/cache/DeviceCacheImplementation.kt

```diff
@@ -23,25 +26,45 @@ class DeviceCacheImplementation @Inject constructor(
     }
 
     override suspend fun getDevice(): Device? {
-        val deviceCacheDto = dataStore.data.firstOrNull()
-        return if (deviceCacheDto == null || deviceCacheDto == DeviceCacheDto()) {
+        val dto = dataStore.data.first()
```

Comment:
This will throw an error if device is null...Unfortunately I think you will need a check for whether the dto is null